### PR TITLE
Remove incorrect __restricted labels to avoid UB.

### PR DESCRIPTION
**Pull request by gAlfonso-bit**
strcspn isn’t supposed to be restricted, and by making it so, it opens the door to UB, as the standard describes the function as finding a complimentary substring, which by definition may include part of the string being compared to.

### CODE REVIEW

**Comment by gAlfonso-bit** [comment]
This issue doesn’t affect the strspn because that doesn’t have the restricted qualifier

**Comment by bsdimp** [comment]
I don't think this is a UB issue. However, __restrict isn't in usermode version of this and it's not in the standard.

File: sys/libkern/strcspn.c [hunk 1]
```diff
@@ -32,10 +32,10 @@ __FBSDID("$FreeBSD$");
 #include <sys/limits.h>
 
 #define	IDX(c)	((u_char)(c) / LONG_BIT)
-#define	BIT(c)	((u_long)1 << ((u_char)(c) % LONG_BIT))
+#define	BIT(c)	(1UL << ((u_char)(c) % LONG_BIT))
```
Comment (bsdimp) [inline sys/libkern/strcspn.c:35]: This is an unrelated change.

**Comment by gAlfonso-bit** [comment]
Fixed.

**Comment by bsdimp** [comment]
https://cgit.freebsd.org/src/commit/?id=56d33e86b74b197a36f42255824b56715c96a596 landed this, closing.
